RecoLocalCalo/HcalRecProducers/plugins/alpaka/Mahi.dev.cc

```diff
@@ -0,0 +1,1457 @@
+
+                // if chi2 is set to -9999 do not run minimization
+                if (outputGPU.chi2()[gch] == -9999.f)
+                  continue;
+
+                // configure shared mem
+                float* shrmem = alpaka::getDynSharedMem<float>(acc);
+                float* shrMatrixLFnnlsStorage = shrmem + calo::multifit::MapSymM<float, NPULSES>::total * lch;
+                float* shrAtAStorage =
+                    shrmem + calo::multifit::MapSymM<float, NPULSES>::total * (lch + threadsPerBlock);
+
+                // conditions for pedestal widths
+                auto const id = gch < f01HEDigis.size()
+                                    ? f01HEDigis.ids()[gch]
+                                    : (gch < nchannelsf015 ? f5HBDigis.ids()[gch - f01HEDigis.size()]
+                                                           : f3HBDigis.ids()[gch - nchannelsf015]);
+                auto const did = DetId{id};
+                auto const hashedId =
+                    did.subdetId() == HcalBarrel
+                        ? did2linearIndexHB(id, mahi.maxDepthHB(), mahi.firstHBRing(), mahi.lastHBRing(), mahi.nEtaHB())
+                        : did2linearIndexHE(id,
+                                            mahi.maxDepthHE(),
+                                            mahi.maxPhiHE(),
+                                            mahi.firstHERing(),
+                                            mahi.lastHERing(),
+                                            mahi.nEtaHE()) +
+                              mahi.offsetForHashes();
+
+                // conditions based on the hash
+                auto const* pedestalWidthsForChannel =
+                    useEffectivePedestals && (gch < f01HEDigis.size() || gch >= nchannelsf015)
+                        ? mahi.effectivePedestalWidths()[hashedId].data()
+                        : mahi.pedestals_width()[hashedId].data();
+                auto const averagePedestalWidth2 = 0.25 * (pedestalWidthsForChannel[0] * pedestalWidthsForChannel[0] +
+                                                           pedestalWidthsForChannel[1] * pedestalWidthsForChannel[1] +
+                                                           pedestalWidthsForChannel[2] * pedestalWidthsForChannel[2] +
+                                                           pedestalWidthsForChannel[3] * pedestalWidthsForChannel[3]);
+
+                // FIXME on cpu ts 0 capid was used - does it make any difference
+                auto const gain = mahi.gains_value()[hashedId][0];
+
+                auto const respCorrection = mahi.respCorrs_values()[hashedId];
+                auto const noisecorr = mahi.sipmPar_auxi2()[hashedId];
+
+#ifdef HCAL_MAHI_GPUDEBUG
+#ifdef HCAL_MAHI_GPUDEBUG_FILTERDETID
+                if (id != DETID_TO_DEBUG)
+                  return;
+#endif
+#endif
+
+                /*
+      // TODO: provide this properly
+      int const soi = soiSamples[gch];
+      */
```

Comment:
is this TODO still relevant ?

Reply:
Unclear, but does not seem relevant anymore. Removed in [3bed955](https://github.com/cms-sw/cmssw/pull/44910/commits/3bed955698f044472ce6896e6dc91d35a17c0393)